Makefile

```
@@ -276,7 +276,7 @@ rvfi_preserve_fns=-c_preserve rvfi_set_instr_packet \
 generated_definitions/c/riscv_rvfi_model_$(ARCH).c: $(SAIL_RVFI_SRCS) model/main.sail Makefile
 	mkdir -p generated_definitions/c
 	$(SAIL) $(rvfi_preserve_fns) $(SAIL_FLAGS) -O -Oconstant_fold -memo_z3 -c -c_include riscv_prelude.h -c_include riscv_platform.h -c_no_main $(SAIL_RVFI_SRCS) model/main.sail -o $(basename $@)
-	sed -i -e '/^[[:space:]]*$$/d' $@
+	sed -i.bak -e '/^[[:space:]]*$$/d' $@
```

```suggestion
	sed -i '' -e '/^[[:space:]]*$$/d' $@
```
`If a zero-length extension is given, no backup will be saved.`

---

I would double check if `-i ''` works with GNU sed, I think it may not `-i''` would without the space I think

---

They're mutually incompatible, unfortunately.

---

Perhaps you could use `perl` to do the same thing? I guess most systems will have it and it might be more portable?

---

```
sed -e '/^[[:space:]]*$$/d' $@ > $@.new
mv $@.new $@
```
is a common portable way to get around the -i importability

---

On second thoughts using sed and having the backups is probably best.

---

@jrtc27 's method is the only true POSIX way... in any case, this discussion has convinced me that a comment is necessary to avoid future pain.

---

I agree.